ariadne_graphql_proxy/contrib/aws/cache_backend.py

```diff
@@ -0,0 +1,66 @@
+import time
+from typing import Any, Optional
+
+from boto3.dynamodb.conditions import Attr, Key
+from boto3.session import Session
+
+from ariadne_graphql_proxy.cache import CacheBackend
+
+
+class DynamoDBCacheError(Exception):
+    """Exception thrown by DynamoDBCacheBackend."""
+
+
+class DynamoDBCacheBackend(CacheBackend):
+    def __init__(
+        self,
+        table_name: str,
+        partition_key: str = "key",
+        ttl_attribute: str = "ttl",
+        session: Optional[Session] = None,
+    ) -> None:
+        self.table_name = table_name
+        self.partition_key = partition_key
+        self.ttl_attribute = ttl_attribute
+        self.value_attribute_name = "value"
+
+        self.session = session if session else Session()
+        self.dynamodb_resource = self.session.resource("dynamodb")
+        self.table = self.dynamodb_resource.Table(self.table_name)
+
+        self._assert_table_exists()
+
+    def _assert_table_exists(self):
+        not_found_exception = (
+            self.dynamodb_resource.meta.client.exceptions.ResourceNotFoundException
+        )
+        try:
+            self.table.table_status
+        except not_found_exception as exc:
+            raise DynamoDBCacheError(
+                f'Unable to access "{self.table_name}" table.'
+            ) from exc
+
+    async def set(self, key: str, value: Any, ttl: Optional[int] = None):
+        item = {self.partition_key: key, self.value_attribute_name: value}
+        if ttl:
+            now = int(time.time())
+            item[self.ttl_attribute] = now + ttl
+
+        self.table.put_item(Item=item)
```

Comment:
I think we should wrap IO operations in asgiref's `sync_to_async` decorators so they don't block the eventloop when cache is accessed.